### PR TITLE
Default exectution policy to AWSLambdaBasicExecutionRole

### PR DESCRIPTION
**Pull request by colincoleman**

### CODE REVIEW

**Review by itsdalmo — CHANGES_REQUESTED** [review]
This won't work; the policy variable is expected to be a JSON policy that can be inlined, so the policy ARN would require a different terraform resource to be attached to the role. And we can't use `count` either, since any use of `${data.aws_iam_policy_document.something.json}` means that the value is computed. And even so, I suspect that the basic execution role still has too wide permissions.

edit: The `terraform-aws-asg` module has the [same problem](https://github.com/telia-oss/terraform-aws-asg/blob/master/variables.tf#L47-L64). And to be honest, I'm inclined to just wait for terraform `0.12.0` to fix this.

**Comment by mikael-lindstrom** [comment]
Good catch on the policy needed to be JSON, missed that. The `arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole` only gives access to write logs to CloudWatch so it seems like a sane default permission if we can solve the JSON part.

**Comment by colincoleman** [comment]
Updated description of of policy to reflect it should be json

File: main.tf [hunk 1]
```diff
@@ -115,8 +115,22 @@ data "aws_iam_policy_document" "assume" {
...
+
+    resources = ["*"]
+  }
+}
+
 resource "aws_iam_role_policy" "main" {
   name   = "${var.name_prefix}-lambda-privileges"
   role   = "${aws_iam_role.main.name}"
-  policy = "${var.policy}"
+  policy = "${var.policy == "" ? data.aws_iam_policy_document.default_lambda.json : var.policy}"
```
Comment (itsdalmo) [inline main.tf:135]: This won't work either. If the value is computed it can't be compared to `""`. 
🐺 🐺🐺🐺🐺

**Pull request closed without merging**